src/main/java/hpi/CopyDataToWorkspacePlugin.java

```diff
@@ -117,25 +131,55 @@ public boolean tearDown(AbstractBuild build, BuildListener listener)
 			}
 		};
 	}
-
-    @Override
-    public Environment setUp(Build build, Launcher launcher, BuildListener listener) throws IOException, InterruptedException {
-        return setUp(build, launcher, listener);
-    }
-
     @Extension
     public static class DescriptorImpl extends BuildWrapperDescriptor {
         public DescriptorImpl() {
             super(CopyDataToWorkspacePlugin.class);
         }
         
         public FormValidation doCheckFolderPath(@AncestorInPath AbstractProject project, @QueryParameter String value) throws IOException {
-            if (0 == value.length()) {
-            	return FormValidation.error("Empty path to folder");
+            if (project != null) {
+                project.checkPermission(Item.CONFIGURE);
             }
-        	return FormValidation.ok();
+            return validateFolderPath(value);
         }
 
+		public static FormValidation validateFolderPath(String value) {
+			if (value == null || value.trim().isEmpty()) {
+				return FormValidation.error("Path cannot be empty");
+			}
+
+			boolean isWindows = System.getProperty("os.name").toLowerCase().contains("windows");

```

Comment:
A method already exists that can do this type of check if you're willing to use it.  `import static hudson.Functions.isWindows;` is needed in the imports section if you choose to apply this change
```suggestion
			boolean isWindows = isWindows();

```